### DISCUSSION TITLE
Android: RecyclerView ListView not updating av.imageview correctly

### DISCUSSION THREAD

**Discussion by m1ga** [appcelerator/titanium_mobile#12557] [Q&A]
Starting with 9.3.0.GA I've noticed that the av.imageview module is not correctly updating images inside a ListView (RecyclerView):

<hr/>

**9.2.2.GA (images show correctly)**

![Screenshot_20210123-211118](https://user-images.githubusercontent.com/4334997/105613393-8452f580-5dc2-11eb-8c7f-767aadb69586.png)
scrolling up, revealing new items
<hr/>

![Screenshot_20210123-211121](https://user-images.githubusercontent.com/4334997/105613394-84eb8c00-5dc2-11eb-8751-4350235c2e97.png)
showing local and remote images

<hr/>

**9.3.1.GA, 9.3.2.GA, 10.x**

![Screenshot_20210123-211302](https://user-images.githubusercontent.com/4334997/105613395-85842280-5dc2-11eb-8ea3-8fb0fc152339.png)
scrolling up, new items are missing
<hr/>

![Screenshot_20210123-211023](https://user-images.githubusercontent.com/4334997/105613391-83ba5f00-5dc2-11eb-81c6-7e499d31b6fb.png)
local images are not visible

<hr/>

**Code**
```xml
<Alloy>
<Window>
	<ListView id="lst" defaultItemTemplate="default">
	    <Templates>
	        <ItemTemplate name="default">
				<ImageView ns="AvImageview" bindId="img" borderColor="red" borderWidth="1" width="128" height="128" />
			</ItemTemplate>
	    </Templates>
	    <ListSection>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	        <ListItem img:image="/images/appicon.png"/>
	    </ListSection>
	</ListView>
</Window>
</Alloy>
```

![ezgif-3-1131831b645f](https://user-images.githubusercontent.com/4334997/105637007-655c6e00-5e6b-11eb-9547-347fbb56822d.gif)

Sometimes it works at the first compile and app start and it will start to ignore the lower images the second time I compile/run the app but then they are gone the whole time.

**Possible sources:**
* Glide in av.imageview
* some caching/memory issue (would explain that it works fine at the first start)
* the RecyclerView somehow has to tell the module to refresh

At the moment I'm not sure if the fix has to be in the SDK inside the RecyclerView or inside the module. And I know that it doesn't happen for everyone but there are other people with the same issue.

**Workaround**
Since I just use that module for the scaleType I've created a custom SDK with https://github.com/appcelerator/titanium_mobile/pull/12285 and use the module only on iOS

**Comment by hansemannn** [comment]
Have you seen https://github.com/appcelerator/titanium_mobile/pull/12541 ? It might help. And we see the same thing, so curious whats happening here.

**Reply by m1ga** [reply]
saw that but not the recent comments. A normal imageview (without scalingType) works fine for me that's why I didn't check that yet. I'll give it a try

**Reply by m1ga** [reply]
* 9.3.3.v20210224080032 - error
* 10.1.0.v20210304082523 - error 
* 10.1.0 master - only had it once but then no image was visible - so might be a connection problem
* 10.1.0 with #12541 - no problems so far :thinking: 

so it looks like something between 2021/03/04 and today fixed it. But have to do some more testing and fix my themes for 10.1.0 :)

**Reply by m1ga** [reply]
there is also this PR now: https://github.com/appcelerator/titanium_mobile/pull/12781
so if someone still had this issue it might be worth a shot. PR description sounds great 😄
